plugin-build/src/main/kotlin/io/sentry/android/gradle/SentryPlugin.kt

```diff
@@ -47,12 +55,23 @@ class SentryPlugin : Plugin<Project> {
                 extraProperties.get(SENTRY_PROJECT_PARAMETER).toString()
             }.getOrNull()
 
+            androidComponentsExtension.onVariants { variant ->
+                variant.transformClassesWith(
+                    SpanAddingClassVisitorFactory::class.java,
+                    InstrumentationScope.ALL
+                ) { params ->
+                    if (extension.forceInstrumentDependencies.get()) {
+                        params.invalidate.setDisallowChanges(System.currentTimeMillis())
+                    }
+                }
+                variant.setAsmFramesComputationMode(FramesComputationMode.COPY_FRAMES)
+            }
```

Comment:
this part should also respect the `ignoredVariants`, `ignoredFlavors` and `ignoredBuildTypes` but we can do it later together with the new APIs of AGP 7

Reply:
I will drop a TODO here and then have a list of checkboxes in the final PR to `main` to resolve this